File: Excel/534 Challenge.py
import pandas as pd

path = "534 Palindrome Numbers with Digit Repeations.xlsx"
input = pd.read_excel(path, usecols="A:B")
test = pd.read_excel(path, usecols="C:E", names=["res1", "res2", "res3"])

def is_palindrome(n):
    return str(n) == str(n)[::-1]

def has_repeated_digits(n, times):
    return any(str(n).count(d) == times for d in str(n))

def find_next_palindromic_numbers(start_number, repeats):
    found_numbers = []
    num = start_number + 1
    while len(found_numbers) < 3:
        if is_palindrome(num) and has_repeated_digits(num, repeats):
            found_numbers.append(num)
        num += 1
    return found_numbers

output = input.apply(lambda x: find_next_palindromic_numbers(x["No. of Digits"], x["Repeats"]), axis=1)
output = output.apply(pd.Series)
output.columns = ["res1", "res2", "res3"]
output = output.astype("int64")

print(output.equals(test))

File: Excel/500-599/534/534 Challenge.R
library(tidyverse)
library(stringi)
library(readxl)

path = "Excel/534 Palindrome Numbers with Digit Repeations.xlsx"
input = read_excel(path, range = "A1:B9")
test  = read_excel(path, range = "C1:E9")

has_digit_repeated_n_times <- function(num, N) {
  any(table(strsplit(as.character(num), "")[[1]]) == N)
}

is_palindrome <- function(num) {
  as.character(num) == stri_reverse(as.character(num))
}

result <- input %>%
  rowwise() %>%
  mutate(
    res = list(
      seq(.data$`No. of Digits`, .data$`No. of Digits` + 1000000) %>% 
        keep(~ is_palindrome(.x)) %>%                                 
        keep(~ has_digit_repeated_n_times(.x, .data$Repeats)) %>%   
        head(3)                                                      
    )
  ) %>%
  unnest(res) %>%
  ungroup()

res <- result %>%
  mutate(nr = row_number(), .by = `No. of Digits`) %>%
  select(-c(Repeats)) %>%
  pivot_wider(names_from = nr, names_glue = "P_{nr}", values_from = res) %>%
  select(-`No. of Digits`)

all.equal(test, res, check.attributes = FALSE)
#> [1] TRUE